src/main/java/it/smartcommunitylab/aac/webauthn/service/WebAuthnRpService.java

```diff
@@ -127,19 +133,21 @@ public WebAuthnRpService(RelyingParty rp,
                 } else {
                     newCred.setTransports("");
                 }
-                newCred.setParentAccount(account);
+                newCred.setParentAccountId(account.getId());
 
-                previousCredentials.add(newCred);
                 webAuthnCredentialsRepository.save(newCred);
-                account.setCredentials(previousCredentials);
-                webAuthnUserAccountRepository.save(account);
                 activeRegistrations.remove(key);
-                return Optional.of(username);
+                if (username != null) {
+                    return username;
+                } else {
+                    throw new WebAuthnAuthenticationException("_",
+                            "Could not register user " + username + " with subject " + account.getSubject());
+                }
             }
         } catch (Exception e) {
-            System.out.println(e);
         }
-        return Optional.empty();
+        throw new WebAuthnAuthenticationException("_",
```

Comment:
this pattern of returning by default an exception is bad. Fix the catch line at 147, at minimum by wrapping the exception message in your custom exception.

Reply:
Done with 5d0907ea035c889865807dd02ec68ab82544ca9d